packages/page-broker/src/Overview/Filters.tsx

```diff
@@ -16,31 +17,31 @@ const StyledDiv = styled.div`
 `;
 
 interface Props {
-  workLoad?: CoreWorkloadType[];
-  onFilter: (data: CoreWorkloadType[]) => void
+  data: CoreInfo[];
+  onFilter: (data: CoreInfo[]) => void
 }
 
-const filterLoad = (parachainId: string, load: CoreWorkloadType[] | CoreWorkplanType[], workloadCoreSelected: number): CoreWorkloadType[] | CoreWorkplanType[] => {
+const filterLoad = (parachainId: string, data: CoreInfo[], workloadCoreSelected: number): CoreInfo[] => {
   if (parachainId) {
-    return load?.filter(({ info }) => info.task === parachainId);
+    return data.filter(({ workload }) => !!workload?.filter(({ info }) => info.task === parachainId).length);
   }
 
   if (workloadCoreSelected === -1) {
-    return load;
+    return data;
   }
 
-  return load?.filter(({ core }) => core === workloadCoreSelected);
+  return data.filter((one) => one.core === workloadCoreSelected);
 };
 
-function Filters ({ onFilter, workLoad }: Props): React.ReactElement<Props> {
+function Filters ({ data, onFilter }: Props): React.ReactElement<Props> {
   const [workloadCoreSelected, setWorkloadCoreSelected] = useState(-1);
   const [_parachainId, setParachainId] = useState<string>('');
 
   const coreArr: number[] = useMemo(() =>
-    workLoad?.length
-      ? Array.from({ length: workLoad?.length || 0 }, (_, index) => index)
+    data?.length
+      ? Array.from({ length: data?.length || 0 }, (_, index) => index)
```

Comment:
```suggestion
      ? Array.from({ length: data.length || 0 }, (_, index) => index)
```
since you are already checking the ternary with `data?.length`